src/CustomParser/Advanced.elm

```diff
@@ -1321,31 +1321,34 @@ isSubChar predicate offset string =
         actualChar =
             String.slice offset (offset + 1) string
     in
-    case actualChar of
-        "" ->
-            -1
+    if charStringIsUtf16HighSurrogate actualChar then
+        -- String.all iterates over code points (so here just one Char)
+        if String.all predicate (String.slice offset (offset + 2) string) then
+            case actualChar of
+                "" ->
```

Comment:
I don't thin `actualChar` can be empty, can it?

Reply:
charStringIsUtf16HighSurrogate does not check for emptiness, it uses `String.all` to be fast

Reply:
I was more thinking in the sense that `String.slice offset (offset + 1) would never be empty, but I guess this could be the end of the file. Alright.